### PR TITLE
Adds InfluxDB Mixin

### PR DESCRIPTION
**Pull request by Caleb-Hurshman**
Adds dashboards and alerts for the InfluxDB mixin.

Dashboards:
- InfluxDB cluster overview
- InfluxDB instance overview
- InfluxDB logs overview

Alerts:
- InfluxDBWarningTaskSchedulerHighFailureRate
- InfluxDBCriticalTaskSchedulerHighFailureRate
- InfluxDBHighBusyWorkerPercentage
- InfluxDBHighHeapMemoryUsage
- InfluxDBHighAverageAPIRequestLatency
- InfluxDBSlowAverageIQLExecutionTime

## InfluxDB cluster overview
influxdb_cluster_overview_1
influxdb_cluster_overview_2
influxdb_cluster_overview_3

## InfluxDB instance overview
influxdb_instance_overview_1
influxdb_instance_overview_2
influxdb_instance_overview_3

## InfluxDB logs overview
influxdb_logs_overview

### CODE REVIEW

**Review by dpaasman00 — CHANGES_REQUESTED** [review]
Few comments, otherwise looks really good. Also be sure to update the pr title before sending over

File: influxdb-mixin/dashboards/influxdb-cluster-overview.libsonnet
Comment (Caleb-Hurshman) [reply influxdb-mixin/dashboards/influxdb-cluster-overview.libsonnet:490]: I don't understand either. Emily is going to provide more direction during the external review.

File: influxdb-mixin/README.md
Comment (dpaasman00) [reply influxdb-mixin/README.md:59]: Oh duh, totally blanked on what I was looking at, good call

File: influxdb-mixin/dashboards/influxdb-cluster-overview.libsonnet
Comment (dpaasman00) [inline influxdb-mixin/dashboards/influxdb-cluster-overview.libsonnet:831]: Notice on this panel the unit is `query/s` where as on the "IQL query rate" panel on the instance dashboard, the unit is "queries/s" despite being essentially the same expression (this one uses topk and the other doesn't). Super nit but I'd say match the unit between these two panels, with whichever one you wanna go with.

**Review by dpaasman00 — COMMENTED** [review]
Looks really good, one last comment

File: influxdb-mixin/alerts/alerts.libsonnet
Comment (BominRahmani) [reply influxdb-mixin/alerts/alerts.libsonnet]: if you really want you can keep it the same since the title differentiates the alert, but it would be better if you can say something like 'Automated data processing tasks are failing at a critically high rate.',

**Review by dpaasman00 — APPROVED** [review]
Looks great

File: influxdb-mixin/dashboards/influxdb-cluster-overview.libsonnet
Comment (StefanKurek) [inline influxdb-mixin/dashboards/influxdb-cluster-overview.libsonnet:1379]: wasn't there a decision to name this something like k8s cluster? or has that not happened yet?

**Review by StefanKurek — COMMENTED** [review]
One small question about the cluster label, but otherwise LGTM

File: influxdb-mixin/dashboards/influxdb-cluster-overview.libsonnet
Comment (Caleb-Hurshman) [reply influxdb-mixin/dashboards/influxdb-cluster-overview.libsonnet:1379]: I never heard any official decision one way or another. I assume it's not a high priority for Grafana.

**Review by StefanKurek — APPROVED** [review]

File: influxdb-mixin/dashboards/influxdb-instance-overview.libsonnet
Comment (Dasomeone) [inline influxdb-mixin/dashboards/influxdb-instance-overview.libsonnet:1863]: Personal take, I'd consider the Go stats quite unimportant and would move them to the bottom of the dashboard.

@rgeyer what are your thoughts here?

File: influxdb-mixin/dashboards/influxdb-cluster-overview.libsonnet
Comment (Dasomeone) [inline influxdb-mixin/dashboards/influxdb-cluster-overview.libsonnet:1425]: Same comment as on the instance dashboard, I'd likely move this row and its panels to the very bottom of the dashboard

File: influxdb-mixin/dashboards/influxdb-instance-overview.libsonnet
Comment (Dasomeone) [inline influxdb-mixin/dashboards/influxdb-instance-overview.libsonnet:966]: I'd suggest suppressing metrics with values = 0, as currently you have this situation where only half the listed metrics are relevant
image

Comment (Dasomeone) [reply influxdb-mixin/dashboards/influxdb-instance-overview.libsonnet:966]: Update: Saw this applied to multiple panels, I'd give that a sanity check throughout if possible.

Basically my fear here is that that's going to multiply with the number of instances, so if it's something like the IQL Query Response Time panel that would probably be fine since it's relevant to see 0s there, but for things like HTTP response it grows out of hand quite quickly.

image

Also think this is a good one to get @rgeyer's thoughts on before actioning

File: influxdb-mixin/README.md
Comment (Dasomeone) [inline influxdb-mixin/README.md:26]: Screenshots haven't been uploaded yet, are the ones in the PR description the ones you'd like us to upload?

File: influxdb-mixin/README.md
Comment (Dasomeone) [inline influxdb-mixin/README.md:41]: Minor nitpick but there is only one screenshot, so no need to call it the first

**Review by Dasomeone — COMMENTED** [review]
Generally looks great, just a few minor comments to discuss

File: influxdb-mixin/README.md
Comment (Caleb-Hurshman) [reply influxdb-mixin/README.md:26]: I was planning to wait until after the mixin was approved to avoid having to re-upload screenshots (eg. after the go panels are potentially moved), but the ones in the PR description are currently the most accurate.

File: influxdb-mixin/dashboards/influxdb-instance-overview.libsonnet
Comment (rgeyer) [reply influxdb-mixin/dashboards/influxdb-instance-overview.libsonnet:1863]: Yes, I'd agree. Go process details are tangentially useful, but should not be prioritized.

File: influxdb-mixin/dashboards/influxdb-instance-overview.libsonnet
Comment (Dasomeone) [reply influxdb-mixin/dashboards/influxdb-instance-overview.libsonnet:966]: That is a very good point re: usability on the metrics changing assigned colours, especially with multiple instances...

Ok yeah then lets exercise caution on this one and maybe just revisit it in the future when we've had time to discuss with Vitaly? Don't want to block the PR

**Review by rgeyer — APPROVED** [review]
LGTM!

File: influxdb-mixin/dashboards/influxdb-instance-overview.libsonnet
Comment (Caleb-Hurshman) [reply influxdb-mixin/dashboards/influxdb-instance-overview.libsonnet]: @Dasomeone @rgeyer Is there a setting in Grafana to suppress 0 values? Nothing I've tried is working quite how I'd like.

Comment (Dasomeone) [reply influxdb-mixin/dashboards/influxdb-instance-overview.libsonnet]: Simply append `> 0` to the end of the query. Should be all you need, though the caveat is that it'll still show up if at any point in the time range it was above 0. Not the end of the world with a properly set time range of 30 minutes or so, but yeah :)

File: influxdb-mixin/README.md
Comment (Dasomeone) [reply influxdb-mixin/README.md:26]: All good, let's just upload the finished ones when we have them then

**Review by Dasomeone — APPROVED** [review]
Looks great, thanks! :D

**Pull request merged**
